android/src/main/java/com/reactnativevolumemanager/VolumeManagerModule.java

```diff
@@ -86,12 +86,14 @@ private void unregisterVolumeReceiver() {
 
   private void setupKeyListener() {
     runOnUiThread(() -> {
+      if (hardwareButtonListenerRegistered) return;
+      if (mContext.getCurrentActivity() == null) return;
```

Comment:
@wilmxre I was facing a similar issue on android keyboard will not close after you edit an input in modal, as you can see in the same method we request focus even if the user is using native ui can we add another check below the check below what you have added to avoid this ?

```suggestion
      if (hardwareButtonListenerRegistered) return;
      if (mContext.getCurrentActivity() == null) return;
      if (showNativeVolumeUI) return;
```

Reply:
if we add this line, i think we should remove the `if (showNativeVolumeUI) return false;` in line 100. what do you think?

Reply:
yes indeed that can be removed it is no longer needed, ideally these check should happen before anything else

Reply:
okay, can you make a change suggestion with that also? i am not close to a computer now. i think we should early return in the cleanupKeyListener method too if the native ui mode is visible

Reply:
yes that would make sense right, if there is no listener we don't have anything to remove, I can't make suggestions for lines that are not changed github has some weird behavior you can't highlights that has not changed